.cursor/rules/responsive-design-patterns.mdc

```diff
@@ -0,0 +1,191 @@
+---
+alwaysApply: true
+---
+
+# Responsive Design: Container Queries Over Media Queries
+
+## Principle
+
+Use container-based styling and `useContainerBreakpoints` for all responsive behavior. Since the SDK is embedded in various contexts and container sizes, component behavior must adapt to the component's actual container, not the viewport. This is essential for predictable behavior across different host applications.
+
+## Guidelines
+
+### Do This ✅
+
+```typescript
+// Good: Use useContainerBreakpoints hook with semantic components
+import { useContainerBreakpoints } from '@/hooks/useContainerBreakpoints'
+import { Flex } from '@/components/Common'
+
+const PayrollReceipts = () => {
+  const breakpoints = useContainerBreakpoints()
+  const isSmall = breakpoints.includes('base') || breakpoints.includes('small')
+
+  return (
+    <Flex
+      flexDirection={isSmall ? 'column' : 'row'}
+      gap={isSmall ? 8 : 16}
+    >
+      {/* Component content */}
+    </Flex>
+  )
+}
+
+// Good: Use semantic Grid component with responsive props
+import { Grid } from '@/components/Common'
+
+const PayrollGrid = () => {
+  const breakpoints = useContainerBreakpoints()
+  const isMobile = breakpoints.includes('base')
+
+  return (
+    <Grid
+      columns={isMobile ? 1 : 2}
+      gap={isMobile ? 8 : 16}
+    >
+      {/* Grid items */}
+    </Grid>
```

Comment:
Grid has responsive props baked in using container queries. We'll want to make it clear that that's the preferred approach instead of reaching for `useContainerBreakpoints`

ex. it should look something like

```
<Grid columns={{ base: 1, small: 2 }} />
```
where the props are passed in as responsive values